### PR TITLE
cleanup feature code after mainnet-beta activation

### PR DESCRIPTION
**Pull request by taozhu-chicago**
#### Problem

feature `EfhYd3SafzGT472tYQDUc4dPd2xdEfKs5fwkowUgVt4W` Remove support for RequestUnitsDeprecated instruction #27500 activated everywhere.

#### Summary of Changes
- Cleanup feature code
- Update/remove tests on that feature gate

Fixes #

### CODE REVIEW

File: sdk/src/compute_budget.rs [hunk 1]
```diff
@@ -23,14 +23,7 @@ crate::declare_id!("ComputeBudget111111111111111111111111111111");
...
 pub enum ComputeBudgetInstruction {
-    /// Deprecated
-    // TODO: after feature remove_deprecated_request_unit_ix::id() is activated, replace it with 'unused'
-    RequestUnitsDeprecated {
-        /// Units to request
-        units: u32,
-        /// Additional fee to add
-        additional_fee: u32,
-    },
+    Unused,
```
Comment (apfitzge) [inline sdk/src/compute_budget.rs:26]: wdyt of putting a small comment here to indicate why it's here.

```rust
    Unused, // deprecated variant, reserved value.
```

File: program-runtime/src/prioritization_fee.rs [hunk 2]
```diff
@@ -5,8 +5,6 @@ type MicroLamports = u128;
 
 pub enum PrioritizationFeeType {
```
Comment (apfitzge) [inline program-runtime/src/prioritization_fee.rs:6]: Do we plan to introduce more variants of prioritization in the near future?
If not, imo we should just get rid of this enum entirely.

**Review by apfitzge — COMMENTED** [review]
Few initial comments, but as you mentioned this will not be merged until future version.

File: program-runtime/src/prioritization_fee.rs (hunk 2 above)
Comment (apfitzge) [reply program-runtime/src/prioritization_fee.rs:6]: maybe just an issue?

Comment (taozhu-chicago) [reply program-runtime/src/prioritization_fee.rs:6]: #34301

**Review by apfitzge — APPROVED** [review]

**Pull request merged**